crates/core/src/factories/core_space.rs

```diff
@@ -312,5 +399,44 @@ impl Space for CoreSpace {
     }
 }
 
+async fn check_agent_infos(
+    config: CoreSpaceConfig,
+    peer_store: peer_store::DynPeerStore,
+    inner: Arc<Mutex<InnerData>>,
+) {
+    loop {
+        // only check at this rate
+        tokio::time::sleep(config.re_sign_freq()).await;
+
+        // only re-sign if they expire within this time
+        let cutoff = Timestamp::now() + config.re_sign_expire_time_ms();
+
+        // get all the local agents
+        let agents = inner
+            .lock()
+            .unwrap()
+            .local_agent_map
+            .values()
+            .cloned()
+            .collect::<Vec<_>>();
+
+        for agent in agents {
+            // is this agent going to expire?
+            let should_re_sign = if let Ok(Some(info)) =
+                peer_store.get(agent.agent().clone()).await
+            {
+                info.expires_at <= cutoff
+            } else {
+                true
```

Comment:
We re-sign on either `Err(_)` or `Ok(None)`, is that okay? I'd expect the error to get logged here at least?

Reply:
Done: https://github.com/holochain/kitsune2/pull/61/commits/9afa6cc40e1072d76614481aa2eeb728a5def2b8